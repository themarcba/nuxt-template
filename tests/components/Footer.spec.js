import {} from 'jest';
// Doc: https://vue-test-utils.vuejs.org/api/
import { createLocalVue, config, shallowMount } from '@vue/test-utils';
import Footer from '@/components/Footer.vue';
import NuxtStubs from '@/tests/nuxt-stubs';

const localVue = createLocalVue();

// Doc: https://vue-test-utils.vuejs.org/api/#config
config.showDeprecationWarnings = true;

describe('Footer.vue', () => {
  const wrapper = shallowMount(Footer, {
    localVue,
    stubs: {
      ...NuxtStubs,
      MbIcon: NuxtStubs.ClientOnly,
      MbLink: NuxtStubs.ClientOnly,
    }
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
