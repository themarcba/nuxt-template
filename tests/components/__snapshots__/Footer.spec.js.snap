// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue matches the snapshot 1`] = `
<footer class="text-secondary text-center py-3">
  Made with <div icon="heart" type="solid"></div> by
  <div href="https://twitter.com/themarcba" inverted="">Marc Backes</div>
</footer>
`;
